sdk/template/Azure.Template/src/Generated/Models/DerivedFromBaseClassWithDiscriminatorUnknown.Serialization.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Text.Json;
+
+namespace Azure.Template.Models
+{
+    internal partial class DerivedFromBaseClassWithUnknownDiscriminator
+    {
+        internal override void WriteCore(Utf8JsonWriter writer)
+        {
+            writer.WriteStartObject();
+            writer.WritePropertyName("discriminatorProperty");
+            writer.WriteStringValue(DiscriminatorProperty);
+            writer.WritePropertyName("baseClassProperty");
+            writer.WriteStringValue(BaseClassProperty);
+            writer.WriteEndObject();
+        }
+
+        internal static DerivedFromBaseClassWithUnknownDiscriminator DeserializeDerivedFromBaseClassWithDiscriminatorUnknown(JsonElement element)
+        {
+            string discriminatorProperty = default;
+            string baseClassProperty = default;
+            foreach (var property in element.EnumerateObject())
```

Comment:
If we want to use this instance in roundtrip scenario, we must preserve all the properties that are available for the element.